src/java.base/share/classes/jdk/internal/classfile/constantpool/Utf8Entry.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.classfile.constantpool;
+
+import jdk.internal.classfile.impl.ConcreteEntry;
+
+/**
+ * Models a {@code CONSTANT_UTF8_info} constant in the constant pool of a
+ * classfile.
+ */
+public sealed interface Utf8Entry
+        extends CharSequence, AnnotationConstantValueEntry
+        permits ConcreteEntry.ConcreteUtf8Entry {
+
+    /**
+     * {@return the string value for this entry}
+     */
+    String stringValue();
+
+    /**
+     * {@return whether this entry describes the same string as the provided string}
+     *
+     * @param s the string to compare to
+     */
+    boolean equalsString(String s);
```

Comment:
Whatif the provided string contains character that are not representable in Utf8? Should we throw (e.g. `IllegalArgumentException`) ?

Reply:
I would rather expect `false` value returned, however this case is probably not covered by tests.

Reply:
You almost got me, I had to think about it for a while :) 
Any String is always representable in Utf8 (or modified Utf8 used in classfile).
It is completely valid to store even 4-byte Unicode characters in Utf8Entry and match them with (Utf16 encoded) String. I'll add such String to the existing test.